### PR TITLE
Do not spawn misbehavior task if it is disabled in config

### PR DESCRIPTION
**Pull request by soareschen**
Closes: #1750

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [x] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by adizere** [comment]
I think we're leaving a crossbeam channel disconnected. This is provoking some issues, specifically: method `spawn_worker_tasks` returns a `cmd_tx`:

https://github.com/informalsystems/ibc-rs/blob/22d362c7249b14cfd81e8a926316801a7679ad1e/relayer/src/worker.rs#L119


The receiver side of `cmd_tx` is created here

https://github.com/informalsystems/ibc-rs/blob/22d362c7249b14cfd81e8a926316801a7679ad1e/relayer/src/worker.rs#L63

In case misbehavior detection is disabled, the receiver side `cmd_rx` is never associated to any task, hence the receiver side gets dropped, while we return the sender side and use it, but this side will be disconnected.

I can see this error by sending an `ft-transfer`, and waiting for a `hermes start` instance to relay that in a separate terminal. The error appearing is:

> 2022-01-10T16:58:01.462321Z ERROR ThreadId(29) [ibc-1] error during batch processing: worker error: error sending through crossbeam channel: sending on a disconnected channel

It seems like this error completely kills the websocket monitor (or supervisor task for that monitor) that handles the subscription to chain `ibc-1`, hence all relaying based on events from this chain freezes, so Hermes never sends the Acknowledgement.

It is weird that our E2E and integration tests both fail to catch this problem.


----

**Complete recipe to reproduce the error I see above:** 

You'll need two chains, and a channel between them.

- terminal 1: `hermes start`, wait for message 'Hermes has started'
- terminal 2: send an `ft-transfer`, e.g.
    - `hermes tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 100 -d samoleans -t 180 -n 1`
- terminal 1: 
    - you should see the successfull relaying `ibc-0 -> ibc-1` of `SendPacket` message
    - but then when chain `ibc-1` produces the ack event for this message, Hermes will fail to receive the event


<details><summary>Complete log</summary>

    Finished dev [unoptimized + debuginfo] target(s) in 0.23s
     Running `target/debug/hermes start`
2022-01-10T17:11:56.915281Z  INFO ThreadId(01) using default configuration from '/Users/adi/.hermes/config.toml'
2022-01-10T17:11:56.918927Z  INFO ThreadId(01) telemetry service running, exposing metrics at http://127.0.0.1:3001/metrics
2022-01-10T17:11:56.918996Z  INFO ThreadId(01) [rest] address not configured, REST server disabled
2022-01-10T17:11:56.936757Z  INFO ThreadId(01) [ibc-0] chain is healthy
2022-01-10T17:11:56.947467Z  INFO ThreadId(01) [ibc-1] chain is healthy
2022-01-10T17:11:56.968205Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-1 -> ibc-0:07-tendermint-0)
2022-01-10T17:11:56.993890Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-1 -> ibc-0:07-tendermint-1)
2022-01-10T17:11:57.018930Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-1 -> ibc-0:07-tendermint-2)
2022-01-10T17:11:57.041745Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-1 -> ibc-0:07-tendermint-3)
2022-01-10T17:11:57.069896Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-0 -> ibc-1:07-tendermint-0)
2022-01-10T17:11:57.094127Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-0 -> ibc-1:07-tendermint-1)
2022-01-10T17:11:57.118914Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-0 -> ibc-1:07-tendermint-2)
2022-01-10T17:11:57.142119Z  INFO ThreadId(01) spawning new background task RefreshClientWorker(ibc-0 -> ibc-1:07-tendermint-3)
2022-01-10T17:11:57.160752Z  INFO ThreadId(01) spawning new background task supervisor_batch
2022-01-10T17:11:57.160825Z  INFO ThreadId(01) spawning new background task supervisor_cmd
2022-01-10T17:11:57.160988Z  INFO ThreadId(01) Hermes has started

2022-01-10T17:12:14.305492Z  INFO ThreadId(29) spawning new background task PacketCmdWorker(ibc-0:transfer/channel-0 -> ibc-1)
2022-01-10T17:12:14.305722Z  INFO ThreadId(29) spawning new background task PacketWorker(ibc-0:transfer/channel-0 -> ibc-1)
2022-01-10T17:12:14.510439Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] generate messages from batch with 1 events
2022-01-10T17:12:14.530185Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] scheduling op. data with 1 msg(s) for Destination (height 0-17612)
2022-01-10T17:12:14.538740Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] relay op. data of 1 msgs(s) to Destination (height 0-17612), delayed by: 8.48862ms [try 1/5]
2022-01-10T17:12:14.538813Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] prepending Destination client update @ height 0-17612
2022-01-10T17:12:14.568682Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] assembled batch of 2 message(s)
2022-01-10T17:12:14.571600Z  INFO ThreadId(18) [ibc-1] refresh: retrieved account sequence=95 number=1
2022-01-10T17:12:14.583557Z  INFO ThreadId(33) [Async~>ibc-1] response(s): 1; Ok:13E1EA700A37F0F62867B8D4702FA5DFCD0EDE47A249F3BCCC4D5CB9C571F8C6

2022-01-10T17:12:14.583610Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] success
2022-01-10T17:12:19.463677Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] clearing pending packets height=Some(Height { revision: 0, height: 17611 })
2022-01-10T17:12:19.490633Z  INFO ThreadId(33) [ibc-0:transfer/channel-0 -> ibc-1] finished scheduling pending packets clearing height=Some(Height { revision: 0, height: 17611 })
2022-01-10T17:12:22.379364Z ERROR ThreadId(29) [ibc-1] error during batch processing: worker error: error sending through crossbeam channel: sending on a disconnected channel



[nothing else happens]
</details>

**Comment by soareschen** [comment]
It looks like in the original design, the code worked even when misbehavior is disabled because the other parts of the client worker keeps the receiver end "alive", even when it is never used. In other words, there was likely memory leaks caused by the worker commands accumulating in the unbounded channel without anything receiving the command.

In the new design, the misbehavior task can quit on its own, and so dangling the sender end of the worker handle and cause errors. I have fixed the logic by making the sending of worker command ignore any sender failure. And if the underlying sender fails, log it and drop the sender. This should be fine as the only error that can arise from a channel sender is disconnect error.

**Review by adizere — APPROVED** [review]
Looks good. I did several manual tests of some corner cases and everything seems neat! Thank you, Soares.

Will let Romain also have a look here, and will merge once his review is through.

Nit: Changelog.

File: relayer/src/worker/handle.rs [hunk 1]
```diff
@@ -1,72 +1,74 @@
...
 
+    pub fn try_send_command(&self, cmd: WorkerCmd) {
+        let res = if let Some(tx) = self.tx.acquire_read().as_ref() {
+            tx.send(cmd)
+        } else {
+            Ok(())
+        };
+
+        if res.is_err() {
+            info!("dropping sender end for worker {} as the receiver is dropped with the worker task terminated", self.id);
```
Comment (romac) [inline relayer/src/worker/handle.rs:48]:
```suggestion
            info!("dropping sender end for worker {} as the receiver was dropped when the worker task terminated", self.id);
```
